corehq/apps/translations/tests/test_bulk_app_translation.py

```diff
@@ -821,6 +821,10 @@ def test_removing_form_translations(self):
             (MODULES_AND_FORMS_SHEET_NAME,
              (('Menu', 'menu1', 'Untitled Module', '', '', '', '', '', '765f110eb62fd26240a6d8bcdccca91b246b96c9'),
               ('Form', 'menu1_form1', 'Untitled Form', '', '', '', '', '', 'fffea2c32b7902a3efcb6b84c94e824820d11856'))),
+              # TODO: This unique ID ^^^ does not match the one in form_modifications/app.json. Why?
+              #
+              # TODO: This value comes from corehq/apps/translations/app_translations/download.py:80
+              #       But fetching the form the same way in corehq/apps/translations/app_translations/upload_app.py:316 fails.
```

Comment:
The fact that the form's unique_id value in the shell is different from its value in the app definition might not matter, as long as `[f.unique_id for m in app.get_modules() for f in m.get_forms()]` is consistent. This is how the values of unique_id are set when the sheet is created in [download.py](https://github.com/dimagi/commcare-hq/blob/68907731201a0ed4e00fceb8b17d249f668ddac3/corehq/apps/translations/app_translations/download.py#L80), so it should find the forms in [upload_app.py](https://github.com/dimagi/commcare-hq/blob/4ad8430a699ae832da83ed1ef0c0144d48c10264/corehq/apps/translations/app_translations/upload_app.py#L316) when the sheet is uploaded. 

But this test is failing because in the test the value of `form.unique_id` *matches* the value in the app definition, i.e. the **app.json** file; it doesn't match the value in line 823 above ... even though the value in line 823 originally comes from `form.unique_id`.

What is happening?!